src/dbt_client/dbtIntegration.ts

```diff
@@ -0,0 +1,317 @@
+import {
+  CancellationToken,
+  ProgressLocation,
+  window,
+  workspace,
+  Disposable,
+} from "vscode";
+import { extendErrorWithSupportLinks, provideSingleton } from "../utils";
+import { PythonBridge, pythonBridge } from "python-bridge";
+import { provide } from "inversify-binding-decorators";
+import {
+  CommandProcessExecution,
+  CommandProcessExecutionFactory,
+} from "../commandProcessExecution";
+import { PythonEnvironment } from "../manifest/pythonEnvironment";
+import { existsSync } from "fs";
+import { TelemetryService } from "../telemetry";
+import { DBTTerminal } from "./dbtTerminal";
+import { ValidateSqlParseErrorResponse } from "../altimate";
+
+export class DBTCommand {
```

Comment:
better to extract interfaces/types into separate types and multiple classes to its own files for improved readability and maintainence

Reply:
I'm not superkeen on having lots of files with very small classes, it's a debatable topic. Separate classes make sense as it would prevent us from passing a wrong type.